llvm/lib/Target/X86/X86ISelDAGToDAG.cpp

```diff
@@ -5305,6 +5339,13 @@ void X86DAGToDAGISel::Select(SDNode *Node) {
       ReplaceNode(Node, CNode);
       return;
     }
+    case Intrinsic::x86_t2rpntlvwz0rs:
+    case Intrinsic::x86_t2rpntlvwz0rst1:
+    case Intrinsic::x86_t2rpntlvwz1rs:
+    case Intrinsic::x86_t2rpntlvwz1rst1:
+      if (!Subtarget->hasAMXTRANSPOSE() || !Subtarget->hasAMXMOVRS())
```

Comment:
Use `!Subtarget->hasAMXMOVRS()` only since we check `!Subtarget->hasAMXTRANSPOSE()` later.